src/main/java/com/google/javascript/cl2dts/DeclarationGenerator.java

```diff
@@ -513,10 +513,32 @@ public Void caseTemplatizedType(TemplatizedType type) {
             emit("[]");
             return null;
           }
+          Iterator<JSType> it = type.getTemplateTypes().iterator();
+          if ("Object".equals(referencedType.getReferenceName())) {
```

Comment:
I think somewhere in the type registry there is a specific constant (marker? something?) that identifies the one, special `Object` type. Maybe you could use that?